packages/hydrogen/src/customer-privacy/ShopifyCustomerPrivacy.tsx

```diff
@@ -143,26 +143,46 @@ export function useCustomerPrivacy(props: CustomerPrivacyApiProps) {
     if (scriptStatus !== 'done' || loadedEvent.current) return;
     loadedEvent.current = true;
 
-    if (!consentConfig.checkoutDomain) logMissingConfig('checkoutDomain');
-    if (!consentConfig.storefrontAccessToken)
-      logMissingConfig('storefrontAccessToken');
+    const {checkoutDomain, storefrontAccessToken} = consentConfig;
+    if (!checkoutDomain) logMissingConfig('checkoutDomain');
+    if (!storefrontAccessToken) logMissingConfig('storefrontAccessToken');
 
     // validate that the storefront access token is not a server API token
     if (
-      consentConfig.storefrontAccessToken.startsWith('shpat_') ||
-      consentConfig.storefrontAccessToken.length !== 32
+      storefrontAccessToken.startsWith('shpat_') ||
+      storefrontAccessToken.length !== 32
     ) {
       // eslint-disable-next-line no-console
       console.error(
         `[h2:error:useCustomerPrivacy] It looks like you passed a private access token, make sure to use the public token`,
       );
     }
 
-    if (withPrivacyBanner && window?.privacyBanner) {
-      window?.privacyBanner?.loadBanner({
-        checkoutRootDomain: consentConfig.checkoutDomain,
-        storefrontAccessToken: consentConfig.storefrontAccessToken,
+    const config: CustomerPrivacyConsentConfig = {
+      checkoutRootDomain: checkoutDomain,
+      storefrontAccessToken,
+    };
+
+    if (checkoutDomain) {
+      let storefrontRootDomain = window.document.location.host;
+      const checkoutDomainParts = checkoutDomain.split('.').reverse();
+      const currentDomainParts = storefrontRootDomain.split('.').reverse();
+      const sameDomainParts: Array<string> = [];
+      checkoutDomainParts.forEach((part, index) => {
+        if (part === currentDomainParts[index]) {
+          sameDomainParts.push(part);
+        }
       });
+
+      storefrontRootDomain = sameDomainParts.reverse().join('.');
+
+      if (storefrontRootDomain) {
+        config.storefrontRootDomain = storefrontRootDomain;
+      }
+    }
+
+    if (withPrivacyBanner && window?.privacyBanner) {
+      window?.privacyBanner?.loadBanner(config);
```

Comment:
Nit picky, but is the conditional property necessary for `window?.`? I don't see it on line 167.